docs(aio): remove `Intl` references

dimakuba opened the pull request:
Since `Intl` API was dropped to use to improve browser support,
would be nice to remove the reference from `Suggested polyfills` as well.

dimakuba committed:
refactor(platform-browser): remove `Intl` references

Since `Intl` API was dropped to use to improve browser support,
would be nice to remove the references of it.

mary-poppins commented:
You can preview 93b10db at https://pr40883-93b10db.ngbuilds.io/.
You can preview 132e790 at https://pr40883-132e790.ngbuilds.io/.

petebacondarwin reviewed (approved):
Good spot @dimakuba.
I think that this is probably really a `refactor(platform-browser)...` type commit that has a docs change with it. Or perhaps split the PR into two PRs one refactoring and one docs.

Otherwise LGTM!

dimakuba commented:
> You can preview [93b10db](https://github.com/angular/angular/commit/93b10db5bee478a2f5fbdcc9dda7feba28abb0e8) at https://pr40883-93b10db.ngbuilds.io/.
> You can preview [132e790](https://github.com/angular/angular/commit/132e7905563a3e301accd4e6cd1397797ec441c2) at https://pr40883-132e790.ngbuilds.io/.

LGTM

dimakuba committed:
docs: remove mention of unused `Intl` polyfill

Since `Intl` API was dropped to use to improve browser support,
it is no longer needed as a polyfill to test Angular.

dimakuba commented:
> Good spot @dimakuba.
> I think that this is probably really a `refactor(platform-browser)...` type commit that has a docs change with it. Or perhaps split the PR into two PRs one refactoring and one docs.
> 
> Otherwise LGTM!

created separate commit

petebacondarwin commented:
Thanks @dimakuba for splitting the commits. Sorry to be a pedant but strictly `refactor(doc-infra)` would be a refactoring of the application that displays the docs. I think we want something like:

```
docs: remove mention of unused `Intl` polyfill

Since `Intl` API was dropped to use to improve browser support,
it is no longer needed as a polyfill to test Angular.
```

mary-poppins commented:
You can preview 2208ff6 at https://pr40883-2208ff6.ngbuilds.io/.

dimakuba commented:
> Thanks @dimakuba for splitting the commits. Sorry to be a pedant but strictly `refactor(doc-infra)` would be a refactoring of the application that displays the docs. I think we want something like:
> 
> ```
> docs: remove mention of unused `Intl` polyfill
> 
> Since `Intl` API was dropped to use to improve browser support,
> it is no longer needed as a polyfill to test Angular.
> ```

done

mary-poppins commented:
You can preview d0502a2 at https://pr40883-d0502a2.ngbuilds.io/.

jelbourn reviewed (approved):
LGTM

Reviewed-for: docs-packaging-and-releasing

AndrewKushnir commented:
[Presubmit](http://test/OCL:358024875:BASE:358024899:1613598619998:fb27d8a8).
